compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1391,6 +1391,11 @@ fn print_native_static_libs(
 ) {
     let mut lib_args: Vec<_> = all_native_libs
         .iter()
+        // Deduplicate consecutive repeated librairies, see rust-lang/rust#113209
+        //
+        // note: we don't use PartialEq/Eq because NativeLib transitivly depends on local
+        // item like spans, which we don't care here and would make the deduplication impossible
+        .dedup_by(|l1, l2| l1.name == l2.name && l1.kind == l2.kind && l1.verbatim == l2.verbatim)
         .filter(|l| relevant_lib(sess, l))
```

Comment:
Deduplication needs to be performed after `relevant_lib` filtering, otherwise you'll have issues like https://github.com/rust-lang/rust/pull/85554.

Reply:
Fixed.